build_tools/python/e2e_model_tests/run_module_utils.py

```diff
@@ -30,6 +30,7 @@ def build_run_flags_for_execution_config(
   """Returns the IREE run module flags of the execution config."""
 
   run_flags = list(module_execution_config.extra_flags)
+  run_flags.append("--device_allocator=caching")
```

Comment:
This feels a bit ad-hoc. Basically we're saying that we *always* run with the caching allocator. I don't think that we want to hardcode this. It should be part of the execution configuration.

Reply:
Yeah, I just had the same thought. I decided to have a `global_flags` in the execution config for this kind of special case. Updated.

Reply:
Found some Python tricks to append the global flags to the extra flags. Not sure if it's easy to read...

Reply:
Ah, nvm, it doesn't allow us to reconstruct the dataclass from dict. Need to fine another solution